src/Symfony/Component/Security/Core/Exception/SafeMessageAuthenticationException.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Exception;
+
+/**
+ * An authentication exception where you can control the message shown to the user.
+ *
+ * Be sure that the message passed to this exception is something that
+ * can be shown safely to your user. In other words, avoid catching
+ * other exceptions and passing their message directly to this class.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class SafeMessageAuthenticationException extends AuthenticationException
+{
+    private $messageKey;
+
+    private $messageData = array();
+
+    public function __construct($message = '', array $messageData = array(), $code = 0, \Exception $previous = null)
+    {
+        parent::__construct($message, $code, $previous);
+
+        if ('' !== $message) {
+            $this->messageKey = $message;
+        }
```

Comment:
hmm, no you removed the complete method call. In such cases, the Symfony standard is to remove any pre-configured defaults in the property definition and just call `$this->messageKey = $message; $this->messageData = $messageData;`